lighthouse-core/audits/critical-request-chains.js

```diff
@@ -40,6 +40,58 @@ class CriticalRequestChains extends Audit {
     };
   }
 
+  static _traverse(tree, cb) {
+    function walk(node, depth, startTime, transferSize = 0) {
+      const children = Object.keys(node);
+      if (children.length === 0) {
+        return;
+      }
+      children.forEach(id => {
+        const child = node[id];
+        if (!startTime) {
+          startTime = child.request.startTime;
+        }
+
+        // Call the callback with the info for this child.
+        cb({
+          depth,
+          id,
+          node: child,
+          chainDuration: (child.request.endTime - startTime) * 1000,
+          chainTransferSize: (transferSize + child.request.transferSize)
+        });
+
+        // Carry on walking.
+        walk(child.children, depth + 1, startTime);
+      }, '');
+    }
+
+    walk(tree, 0);
+  }
+
+  /**
+   * Get stats about the longest initiator chain (as determined by time duration)
+   * @return {{duration: number, length: number, transferSize: number}}
+   */
+  static _getLongestChain(tree) {
```

Comment:
move from the formatter to the audit. I think it was only in the formatter because we didn't have `extendedInfo` when this was first written